metadata/db.go

```diff
@@ -325,16 +332,18 @@ func writeMetadataEntry(md *bolt.Bucket, m *metadataEntry) error {
 			firstChildName, firstChild = name, child
 			break
 		}
-		if err := md.Put(bucketKeyChildID, encodeID(firstChild.id)); err != nil {
-			return fmt.Errorf("failed to put id of first child %q: %w", firstChildName, err)
-		}
-		if err := md.Put(bucketKeyChildName, []byte(firstChildName)); err != nil {
-			return fmt.Errorf("failed to put name first child %q: %w", firstChildName, err)
-		}
 		if len(m.children) > 1 {
+			// Sort children by base name.
+			keys := make([]string, len(m.children))
+			for k := range m.children {
+				keys = append(keys, k)
+			}
```

Comment:
If we want to preallocate, then it should be `keys := make([]string, 0, len(m.children)`. Right now we are using twice the memory.

The current form will create the container with size len(m.children) and fill with zero values. i.e. empty strings. This results in the empty keys check below. That can be removed with this fix.

Reply:
Ah, right. I forgot the capacity is doubled once we hit the length.